polyfactory/factories/base.py

```diff
@@ -1038,8 +1050,15 @@ def process_kwargs(cls, **kwargs: Any) -> dict[str, Any]:
         for field_meta in cls.get_model_fields():
             field_build_parameters = cls.extract_field_build_parameters(field_meta=field_meta, build_args=kwargs)
             if cls.should_set_field_value(field_meta, **kwargs) and not cls.should_use_default_value(field_meta):
-                if hasattr(cls, field_meta.name) and not hasattr(BaseFactory, field_meta.name):
-                    field_value = getattr(cls, field_meta.name)
+                has_field_value = hasattr(cls, field_meta.name)
+                field_value = has_field_value and getattr(cls, field_meta.name)
+
+                # NeverNone & AlwaysNone should be treated as a normally-generated field, since this changes logic
+                # within get_field_value.
+                excluded_field_value = has_field_value and isinstance(field_value, (NeverNone, AlwaysNone))
+
+                # TODO why do we need the BaseFactory check here, only dunder methods which are ignored would trigger this?  # noqa: FIX002
```

Comment:
```suggestion
                # Avoid considering fields on base field in case of clash
```
or if fine to remove check then can do that